test/parallel/test-child-process-kill.js

```diff
@@ -60,3 +60,12 @@ if (common.isWindows) {
   });
   process.kill('SIGHUP');
 }
+
+// Test that the process is not killed when sending a 0 signal.
+// This is a no-op signal that is used to check if the process is alive.
+const checkProcess = spawn(process.execPath, ['-e', 'setTimeout(() => {}, 1000)']);
+checkProcess.on('exit', (code, signal) => {
+  assert.strictEqual(code, 0);
+  assert.strictEqual(signal, null);
+});
+checkProcess.kill(0);
```

Comment:
```suggestion
const checkProcess = spawn(process.execPath, [
  '-e',
  'setInterval(() => {}, 1000)'
]);

checkProcess.on('exit', (code, signal) => {
  assert.strictEqual(code, null);
  assert.strictEqual(signal, 'SIGTERM');
});

checkProcess.kill(0);  // 'SIGKILL' is not sent to the child process.
checkProcess.kill('SIGTERM');
```

Just a suggestion to not rely on timers and not wait a second for the test to end.

Reply:
I like changing timeout to interval and then killing it manually to have complete control. However, the new asserts change the nature of the test itself. We want to test `checkProcess.kill(0);` and see if the code and signal will be as expected.  This change checks if sending `0` will kill the process, but not if it'll return the expected values. Correct?

Reply:
Yes, correct, the assumption is that `checkProcess.kill(0)` is a no-op because the received signal is `'SIGTERM'`. If it wasn't the received signal would be `'SIGKILL'`. I think this is sufficient to assert that it doesn't change the signal. However, as you said, there is no check on the exit code.

Reply:
Another option can be something like this:

```js
const code = `const interval = setInterval(() => {}, 1000);
process.stdin.on('data', () => { clearInterval(interval); });
process.stdout.write('x');
`;

const checkProcess = spawn(process.execPath, ['-e', code]);

checkProcess.on('exit', (code, signal) => {
  assert.strictEqual(code, 0);
  assert.strictEqual(signal, null);
});

checkProcess.stdout.on('data', common.mustCall((chunk) => {
  assert.strictEqual(chunk.toString(), 'x');
  checkProcess.kill(0);
  checkProcess.stdin.write('x');
  checkProcess.stdin.end();
}));
```

What do you think?

Reply:
I like that one. It's more elegant IMHO. You can submit it as a new suggestion if you want.

Reply:
Feel free to apply it yourself.